src/EVM.hs

```diff
@@ -1306,29 +1304,33 @@ accessStorage
   -> (Expr EWord -> EVM s ())
   -> EVM s ()
 accessStorage addr slot continue = do
+  let slotConc = Expr.concKeccakSimpExpr True slot
   use (#env % #contracts % at addr) >>= \case
     Just c ->
+      -- Try first without concretization. Then if we get a Just, with concretization
+      -- if both give a Just, should we `continue`
```

Comment:
this does not spark joy, but I think we need a larger refactor of the rpc stuff to handle this properly (out of scope for this pr imo).

Reply:
Good point. I have now explained this in more detail!